src/main/java/io/localmotion/chatbox/notification/ChatBoxNotificationRepository.java

```diff
@@ -0,0 +1,70 @@
+package io.localmotion.chatbox.notification;
+
+import io.micronaut.configuration.hibernate.jpa.scope.CurrentSession;
+import io.micronaut.spring.tx.annotation.Transactional;
+import org.springframework.transaction.annotation.Propagation;
+
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+@Singleton
+public class ChatBoxNotificationRepository implements IChatBoxNotificationRepository {
+
+	@PersistenceContext
+	private EntityManager entityManager;
+
+	public ChatBoxNotificationRepository(@CurrentSession EntityManager entityManager) {
+		this.entityManager = entityManager;
+	}
+
+	public static final String selectDistinctUsers = "select distinct m.chatBoxNotificationId.personId from ChatBoxNotification m ";
+	@Override
+	@Transactional(readOnly = true)
+	public List<String> findDistinctPersons() {
+		return entityManager.createQuery(selectDistinctUsers,String.class).getResultList();
+	}
+
+	public static final String selectAllChatBoxesByPerson = "select m from ChatBoxNotification m " +
+															"where m.chatBoxNotificationId.personId = :personId" ;
+	@Override
+	@Transactional(readOnly = true)
+	public List<ChatBoxNotification> findAllChatBoxesByPerson(String person) {
+		return entityManager.createQuery(selectAllChatBoxesByPerson, ChatBoxNotification.class)
+				.setParameter("personId", person)
+				.getResultList();
+	}
+
+	public static final String selectLastReadMessageIdByPersonandChatBox = "select m from ChatBoxNotification m " +
+																			"where m.chatBoxNotificationId.personId = :personId AND m.chatBoxNotificationId.chatBoxId = :chatBoxId";
+	@Override
+	@Transactional(readOnly = true)
+	public ChatBoxNotification findReadMessageIdByPersonAndChatBox(String person, String chatBox) {
```

Comment:
ChatBox -> Chatbox

Reply:
I have made the changes 
ChatBox -> Chatbox